steamtinkerlaunch

```diff
@@ -25923,7 +25928,13 @@ function checkSteamDeckDependencies {
 		local CHECKCMD
 		CHECKCMD="$($DEPCMD &> /dev/null --version && echo "OK" || echo "NOK")"
 
-		if [ -f "$(command -v "$DEPCMD")" ] && [ "$CHECKCMD" = "OK" ]; then
+
+		# The check now says, don't update dependencies if all of these conditions are true:
+		#  1. The dependency file exists
+		#  2. The dependency can actually be ran, confirming it is a valid file
+		#  3. SteamTinkerLaunch has not updated or autoupdater isn't enabled, so there would be no change in dependency version and thus no need to update
+		# If any of these are false, we need to check our dependencies (if a file is missing we would need to update, or if it cannot be used we need to update, and also if STL updated we may need a newer version, so update).
+		if [ -f "$(command -v "$DEPCMD")" ] && [ "$CHECKCMD" = "OK" ] && { [ "$( checkSteamDeckSTLUpdated )" -eq 0 ] || [ "$( STEAMDECK_AUTOUP )" -eq 0 ] }; then
```

Comment:
ShellCheck is angry about [SC1083](https://www.shellcheck.net/wiki/SC1083). This is the `if` check for checking if we need to update a dependency. `checkSteamDeckSTLUpdated` checks if the last version and current version match.

I guess what we're trying to do here is a "compound" check, right? At least in my head the check is supposed to be like this in other languages that have much saner syntax than Bash: `if -f DEPCMD and CHECKCMD == "OK" and ( checkSteamDeckSTLUpdated returns 0 and STEAMDECK_AUTOUP == 0 )`. The last two "terms" so to speak are treated as one conditional, since we only care if the STL version changed when auto-updates are enabled.

Compound checking like this in Bash is something that constantly escapes me and I have to keep looking it up in the code. But the following should work: `if [[ ( "$( checkSteamDeckSTLUpdated )" -eq 0 && "$STEAMDECK_AUTOUP" -eq 1 ) && -f "$(command -v "$DEPCMD")" && "$CHECKCMD" = "OK" ]]; then`. I validated that ShellCheck at least doesn't get angry anymore :-)

Note that I changed the `STEAMDECK_AUTOUP` check.
- Before, it was like this: `"$( STEAMDECK_AUTOUP )" -eq 0`
    - This may work, ShellCheck didn't complain about it at least.
- However using `"$( FOOBAR )"` is typically to capture function returns. When you just want to check a variable, you can do `[ "$FOOBAR" -eq 0 ]`.
    - I used the `[]` here because that's typically how you do a test, but when doing compound checks you have to use the funky `[[]]` and put your grouped terms in `()`. For example `[[ ( foo && bar ) || ( fizz && buzz ) && foobar && fizzbuzz ]]`.

At least, that is my understanding, and I'm certainly not a Bash expert! But I hope I helped clarify a little bit about the suggestions and where they came from :smile:

Reply:
I just fixed it, thanks a lot for the whole explanation about bash scripting 

> I also changed it to an -eq 1 check, because we want to make sure the value is enabled.

It still should be `( "$( checkSteamDeckSTLUpdated )" -eq 0 || "$STEAMDECK_AUTOUP" -eq 0 )`, we don't want to update if either `lastver==current_ver` **OR** the autoupdater isn't enabled (since it's an OR equation it will bypass the `lastver==current_ver` when it mismatches)

Reply:
ShellCheck is happy now, and your clarification makes sense (resolved and forgot to leave the reply, apologies).